crates/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -42,29 +42,48 @@ pub struct Transaction {
     /// Block hash
     pub block_hash: Option<B256>,
     /// Block number
-    pub block_number: Option<U256>,
+    #[serde(with = "alloy_serde::num::u64_hex_opt")]
+    pub block_number: Option<u64>,
     /// Transaction Index
-    pub transaction_index: Option<U256>,
+    #[serde(with = "alloy_serde::num::u64_hex_opt")]
+    pub transaction_index: Option<u64>,
     /// Sender
     pub from: Address,
     /// Recipient
     pub to: Option<Address>,
     /// Transferred value
     pub value: U256,
     /// Gas Price
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub gas_price: Option<U256>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "alloy_serde::num::u128_hex_or_decimal_opt"
```

Comment:
same question. pretty sure decimal would be illegal rpc response here? I don't mind tolerating it, but @mattsse asked for removal of decimal in other PRs

Reply:
+ 1 to removing decimal support

Reply:
makes sense, let's enforce strict quantity rules everywhere for core eth types

Reply:
side note, the `U***` type deser DO accept decimal. so this is technically a breaking behavior change